### ISSUE TITLE
install fails on arm(beaglebone black) Debian GNU/Linux 8 

### ISSUE THREAD

**Issue by CTCHunter1** [dturing/node-gstreamer-superficial#3]
installing with the bellow command fails on the beaglebone black. I'm suspecting it's from an unlisted dependency.

bash dump: 
root@bone:/var/lib/cloud9/Projects/VideoServer/node-gstreamer-superficial# npm install @skyrising/gstreamer-superficial --save
> @skyrising/gstreamer-superficial@1.0.2 install /var/lib/cloud9/Projects/VideoServer/node-gstreamer-superficial/node_modules/@skyrising/gstreamer-superficial
> node-gyp rebuild

module.js:338
    throw err;
          ^
Error: Cannot find module 'nan'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at [eval]:1:1
    at Object.exports.runInThisContext (vm.js:74:17)
    at Object.<anonymous> ([eval]-wrapper:6:22)
    at Module._compile (module.js:460:26)
    at evalScript (node.js:431:25)
    at startup (node.js:90:7)
gyp: Call to 'node -e "require('nan')"' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:305:16)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1078:12)
gyp ERR! System Linux 4.4.9-ti-r25
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /var/lib/cloud9/Projects/VideoServer/node-gstreamer-superficial/node_modules/@skyrising/gstreamer-superficial
gyp ERR! node -v v0.12.17
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok 
npm WARN gstreamer-superficial@0.0.2 No license field.
npm ERR! Linux 4.4.9-ti-r25
npm ERR! argv "node" "/usr/local/bin/npm" "install" "@skyrising/gstreamer-superficial" "--save"
npm ERR! node v0.12.17
npm ERR! npm  v3.8.2
npm ERR! code ELIFECYCLE

npm ERR! @skyrising/gstreamer-superficial@1.0.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @skyrising/gstreamer-superficial@1.0.2 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @skyrising/gstreamer-superficial package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @skyrising/gstreamer-superficial
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @skyrising/gstreamer-superficial
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /var/lib/cloud9/Projects/VideoServer/node-gstreamer-superficial/npm-debug.log

[npm-debug.log.txt](https://github.com/dturing/node-gstreamer-superficial/files/691674/npm-debug.log.txt)

**Comment by tvandergeer** [comment]
Did you try ```npm install nan``` ?

**Comment by dturing** [comment]
I've added nan as a proper dependency now (it was only a development dependency). This seems to help.

@CTCHunter1 does that help?

**Comment by CTCHunter1** [comment]
I tried to install it again though NPM. It's still throwing a cannot find nan package error.  Maybe the git changes aren't in NPM yet. 

# npm update
root@cytobone:/var/lib/cloud9/Projects/Test# npm install @skyrising/gstreamer-superficial --save

> @skyrising/gstreamer-superficial@1.0.2 install /var/lib/cloud9/node_modules/@skyrising/gstreamer-superficial
> node-gyp rebuild

module.js:338
    throw err;
          ^
Error: Cannot find module 'nan'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at [eval]:1:1
    at Object.exports.runInThisContext (vm.js:74:17)
    at Object.<anonymous> ([eval]-wrapper:6:22)
    at Module._compile (module.js:460:26)
    at evalScript (node.js:431:25)
    at startup (node.js:90:7)
gyp: Call to 'node -e "require('nan')"' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:305:16)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1078:12)
gyp ERR! System Linux 4.4.9-ti-r25
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /var/lib/cloud9/node_modules/@skyrising/gstreamer-superficial
gyp ERR! node -v v0.12.18
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok 
/var/lib/cloud9
├── express@4.14.0 
└── update@0.7.4 

npm WARN enoent ENOENT, open '/var/lib/cloud9/package.json'
npm WARN cloud9 No description
npm WARN cloud9 No repository field.
npm WARN cloud9 No README data
npm WARN cloud9 No license field.
npm ERR! Linux 4.4.9-ti-r25
npm ERR! argv "node" "/usr/local/bin/npm" "install" "@skyrising/gstreamer-superficial" "--save"
npm ERR! node v0.12.18
npm ERR! npm  v3.8.2
npm ERR! code ELIFECYCLE

npm ERR! @skyrising/gstreamer-superficial@1.0.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @skyrising/gstreamer-superficial@1.0.2 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @skyrising/gstreamer-superficial package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @skyrising/gstreamer-superficial
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @skyrising/gstreamer-superficial
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /var/lib/cloud9/Projects/Test/npm-debug.log

**Comment by dturing** [comment]
i guess any project would ask you to use the latest sources when reporting issues, so here ya go:

* if you have an issue with @skyrising/gstreamer-superficial (from npm), post your issue at [the matching project](https://github.com/skyrising/node-gstreamer-superficial),
* for reporting issues with dturing/node-gstreamer-superficial, please use the code from that exact project. there is no "official" npm package (yet), so use from git.

i'm closing this issue, please continue in #4

**Comment by CTCHunter1** [comment]
by latest sources I did an apt-get update and apt-get ugrade in Debian stable. That put me at version 0.12. The latest node release is 7.4.0. 0.12 is still supported in Deiban stable but is old.

If your running Debian (stable) and looking to get this to work, I resolved this by installing the latest version of node with [nvm](https://github.com/creationix/nvm). It brought me to version 7. The downside is now nvm and node are installed outside of the package manager.
